helpers/auth/authentication.py

```diff
@@ -128,17 +126,20 @@ def wrapper(*args, **kwargs):
                     else:
                         user.location = "Nairobi"
                         user.save()
-                    user_role = UsersRole.query.filter_by(
-                        user_id=user.id).first()
-                    role = Role.query.filter_by(id=user_role.role_id).first()
 
-                    if role.role in expected_args:
-                        return func(*args, **kwargs)
+                    if user.roles:
+                        role = user.roles[0]
+                        if role.role in expected_args:
+                            return func(*args, **kwargs)
+                        else:
+                            message = (
+                                'You are not authorized to perform this action')
+                            status = 401
+                            handle_http_error(message, status, expected_args)
                     else:
-                        message = (
+                        raise GraphQLError(
```

Comment:
@dnuwa why did you remove the http error handler here. I believe it is meant to help you raise these such GraphQL errors.